node/src/config_files/mod.rs

```diff
@@ -40,58 +35,62 @@ use self::{
 };
 
 /// The node configuration.
+#[must_use]
 #[derive(Serialize, Deserialize, Debug)]
 pub struct NodeConfigFile {
-    /// The path to the data directory.
-    ///
-    /// By default the config file is created inside of the data directory.
-    pub datadir: PathBuf,
-
     // Subsystems configurations.
-    pub chainstate: ChainstateLauncherConfigFile,
-    pub p2p: P2pConfigFile,
-    pub rpc: RpcConfigFile,
+    pub chainstate: Option<ChainstateLauncherConfigFile>,
+    pub p2p: Option<P2pConfigFile>,
+    pub rpc: Option<RpcConfigFile>,
 }
 
 impl NodeConfigFile {
     /// Creates a new `Config` instance with the given data directory path.
```

Comment:
I think this comment should be removed or updated because the data directory is no longer stored in the config file.

Reply:
Done